### PR TITLE
Add unlockAndSlash function

### PR DESCRIPTION
**Pull request by bingen**
Rebase and merge after #16

### CODE REVIEW

File: contracts/Staking.sol [hunk 1]
```diff
@@ -157,6 +157,28 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
         _decreaseActiveLockAmount(_from, msg.sender, _amount);
     }
 
+    /**
+     * @notice Transfer `_amount` tokens from `_from`'s lock by `msg.sender` to `_to`, and unlock the rest
```
Comment (facuspagnuolo) [inline contracts/Staking.sol:161]: Let's use the token amount notation of radspec here

File: test/locking.js [hunk 2]
```diff
@@ -287,6 +287,25 @@ contract('Staking app, Locking', ([owner, user1, user2]) => {
...
+
+    assert.equal((await staking.unlockedBalanceOf(owner)).toString(), (totalLock - slashAmount).toString(), "Unlocked balance should match")
+    assert.equal((await staking.getTotalLockedOf(owner)).toString(), '0', "total locked doesn’t match")
+    // lock manager
+    assert.equal((await staking.unlockedBalanceOf(user1)).toString(), '0', "Unlocked balance should match")
+    assert.equal((await staking.getTotalLockedOf(user1)).toString(), '0', "total locked doesn’t match")
+    // recipient
+    assert.equal((await staking.unlockedBalanceOf(user2)).toString(), slashAmount.toString(), "Unlocked balance should match")
+    assert.equal((await staking.getTotalLockedOf(user2)).toString(), '0', "total locked doesn’t match")
+  })
```
Comment (facuspagnuolo) [inline test/locking.js:307]: Let's add another test to check this function cannot be called by a non-manager account

File: contracts/Staking.sol [hunk 3]
```diff
@@ -157,6 +157,28 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
...
+    )
+        external
+        isInitialized
+    {
+        // No need to check that amount is positive, as _transfer will fail
+        // No need to check that have enough locked funds, as _decreaseActiveLockAmount will fail
+
+        _transfer(_from, msg.sender, _to, address(0), _amount);
+        _decreaseActiveLockAmount(_from, msg.sender, _amount);
+        _unsafeUnlock(_from, msg.sender);
```
Comment (facuspagnuolo) [inline contracts/Staking.sol:179]: I haven't analyzed if a reentrance is possible, but it feels safer to do the transfer at the end

**Comment by facuspagnuolo** [comment]
I think we should rename this method to `reduceAndTransferFromLock` to keep it consistent with the current terminology

File: contracts/Staking.sol (hunk 3 above)
Comment (bingen) [reply contracts/Staking.sol]: But there’s no real transfer here, I mean, we are not calling any `token.transfer` (nor any other contract actually).

**Pull request merged**